### DISCUSSION TITLE
Prescribing value on a set of nodes

### DISCUSSION THREAD

**Discussion by tairoon1** [idaholab/moose#19079] [Q&A General] [answered]
Hi,

I was wondering, is it possible to prescribe the values of a variable on an edge not by selecting the edge by name and then giving a value or function, but actually to use a node list and corresponding value list for each node? So something like:
```
[BCs]
  [prescribed]
    type = DirichletBC
    variable = u
    node_ids = '0 1 2 3 4'
    values = '0 0.1 0.2 0.3 0.4'
  []
[]
```
Thanks for your help.

**Comment by ngrilli** [comment, marked as answer]
Dear @tairoon1 

You need to look into the MeshGenerators,
normally I use this one:
https://github.com/idaholab/moose/blob/next/framework/src/meshgenerators/SideSetsFromBoundingBoxGenerator.C

in which I set the coordinates for a box, and the nodes inside are added to a node set.

I cannot see any MeshGenerator in the framework that can work with the node ids, but should not be too difficult to make,
you should also check in the modules.

I am quite sure the solution will not be to add the ids to a BC, but using a MeshGenerator.

Best Regards,
Nicolò

**Reply by GiudGiud** [reply]
Docs for the one @ngrilli suggested, I would do it that way too
https://mooseframework.inl.gov/source/meshgenerators/SubdomainBoundingBoxGenerator.html

~~This one works with ids but you need to specify the ids for ALL elements
https://mooseframework.inl.gov/source/meshgenerators/SubdomainIDGenerator.html~~
EDIT : that last one is for block ids. We could make a new one for sideset IDs though

**Reply by tairoon1** [reply]
Thank you for the replies. If I understand correctly, then the MeshGenerators allows you to select a node set. Assuming now, I have that node set, how could I then prescribe values onto each degree of freedom of each node?

**Reply by ngrilli** [reply]
Dear @tairoon1 

A MeshGenerator object always provides a name for the newly created node set.
You can use that name in the boundary conditions.

You can see an example here:
https://github.com/ngrilli/c_pfor_am/blob/main/test/tests/neper/polycrystalneper.i
Note the line: 
"new_boundary = x0"

Best Regards,
Nicolò

**Reply by GiudGiud** [reply]
So you'll have to create a DirichletBC for every nodeset you created. As long as there s like less than 1000 it should be fine speedwise.
If there are a lot of nodes you want to specify, you may want to either:
- write a script to write the input file
- write an Action (in moose, in C++) to make it create the nodesets and the dirichlet BC from just a vector

**Reply by tairoon1** [reply]
I see, thank you for your replies. I think I understand now how it works. I was hoping that it would be possible to directly specify directly given a list of node_ids/dof_ids and then a list of values, but I guess that would be quite specific. Thanks for your help!
